src/test/ui/issue-27282-move-match-input-into-guard.stderr

```diff
@@ -0,0 +1,32 @@
+error[E0505]: cannot move out of `b` because it is borrowed
+  --> $DIR/issue-27282-move-match-input-into-guard.rs:26:16
+   |
+LL |        match b {
+   |   _____-
+   |  |_____|
+   | ||
+LL | ||         &mut false => {},
+LL | ||         _ if { (|| { let bar = b; *bar = false; })();
+   | ||                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ move out of `b` occurs here
+LL | ||                      //~^ ERROR cannot move out of `b` because it is borrowed [E0505]
+...  ||
+LL | ||         _ => panic!("surely we could never get here, since rustc warns it is unreachable."),
+LL | ||     }
+   | ||     -
+   | ||_____|
+   | |______borrow of `b` occurs here
+   |        borrow later used here
```

Comment:
_Note to self, dig for other cases of diagnostics emitting the same span and rework the emitter to just use one span with multiple labels_

Reply:
Also, the AST based errors avoid emitting the two labels if they end up in the same span, NLL should do the same.